Sources/APPX.cpp

```diff
@@ -102,6 +104,15 @@ namespace appx {
             for (const auto &fileNamePair : fileNames) {
                 const std::string &archiveName = fileNamePair.first;
                 const std::string &fileName = fileNamePair.second;
+
+                const std::string suffix = "AppxBundleManifest.xml";
+                if (bundle &&
+                        suffix.size() < archiveName.size() &&
+                        std::equal(suffix.rbegin(), suffix.rend(), archiveName.rbegin())) {
```

Comment:
> We are doing the same thing with AppxManifest.xml for .APPX archives.

Are we? I grepped the code for "AppxManifest" and nothing came up.

> Basically, these two are the only files with hardcoded names that we need to provide externally (i.e. that we do not generate by the appx executable)

AppxManifest.xml isn't required for .appx files. (It is required for Windows 10 apps though.) I don't recall off-hand when you'd want this though. =S

AppxBundleManifest.xml should be generated by fb-util-for-appx, not supplied by the user. I understand for complexity reasons that isn't done yet. Because AppxBundleManifest.xml is special, it shouldn't be in the normal input file list. Because fb-util-for-appx doesn't generate this file itself, the user must supply it somehow. But because it's special, the user shouldn't treat it as a normal file, and fb-util-for-appx shouldn't make it look like a normal file.

It should be impossible-by-usage for the user to make a bundle *without* AppxBundleManifest.xml.